### DISCUSSION TITLE
Challenges in Establishing a Stable Connection: Building the srsRAN gNB with srsUE

### DISCUSSION THREAD

**Discussion by rajugbv** [srsran/srsRAN_Project#348] [General Help] [answered]
Hi everyone, I hope everyone is doing well.
I am trying to build the 'srsRAN gNB with srsUE project.' For the gNB and UE, I am using USRP-2901. I was able to successfully establish a connection between the gNB and UE  but it is not stable

 I was able to transfer video using an SMA cable, as well as an antenna. For the SMA cable, I managed to send videos ranging from 30 seconds to 5 minutes in various tries. With the antenna, I successfully sent videos from 5 seconds to 30 seconds in different attempts.

However, I encountered a problem. For each new try, I have to restart the UE side laptop. I have tried removing the IP address, but it didn't work. Ping is functioning correctly, but I am unable to get output from the 'iperf3' command. Perhaps it is not getting enough time to execute.
[gnb_file.zip](https://github.com/srsran/srsRAN_Project/files/13530963/gnb_file.zip)
for ue side file please check the link below
https://drive.google.com/file/d/1HflyrCT6CrIKjlgN1GVWgr8St3hU8ixD/view?usp=drive_link

**Comment by herlesupreeth** [comment, marked as answer]
There are couple of things I notice in you configuration file and logs

1. Please use the latest code from this repo and UE repo. If this is a bug it may have been fixed in latest code (Dont forget to recompile after fetching latest code)
2. When using the latest code the configuration file for gNB need to be updated - please refer to the documentation again as its updated
3. Execute the performance script - https://github.com/srsran/srsRAN_Project/blob/main/scripts/srsran_performance
4. I would suggest to change the gain values as follows and give it a try
```
  tx_gain: 80                                                     # Transmit gain of the RF might need to adjusted to the given situation.
  rx_gain: 40                                                     # Receive gain of the RF might need to adjusted to the given situation. 
```
